graphicsmagick/src/utils/magick_alloc.rs

```diff
@@ -115,13 +117,15 @@ impl<T> MagickBoxSlice<T> {
     /// of U.
     /// T must be able to deal with all valid representation of U.
     pub(crate) unsafe fn new<U>(a: *mut U, len: usize) -> Option<Self> {
-        assert_eq!(mem::size_of::<U>(), mem::size_of::<T>());
+        unsafe {
+            assert_eq!(mem::size_of::<U>(), mem::size_of::<T>());
```

Comment:
```suggestion
            const { assert_eq!(mem::size_of::<U>(), mem::size_of::<T>()) };
```

By `const`ifying it, we would get compile failure instead of a runtime error